src/routes/notifications/v1/notifications.controller.ts

```diff
@@ -156,20 +217,52 @@ export class NotificationsController {
     @Param('safeAddress', new ValidationPipe(AddressSchema))
     safeAddress: `0x${string}`,
   ): Promise<void> {
+    if (this.isPushNotificationV2Enabled) {
+      return this.unregisterSafeV2Compatible(chainId, uuid, safeAddress);
+    }
+
     await this.notificationsService.unregisterSafe({
       chainId,
       uuid,
       safeAddress,
     });
+  }
 
-    if (this.isPushNotificationV2Enabled) {
+  private async unregisterSafeV2Compatible(
+    chainId: string,
+    uuid: UUID,
+    safeAddress: `0x${string}`,
+  ): Promise<void> {
+    try {
       // Compatibility with V2
-      // @TODO Remove NotificationModuleV2 after all clients have migrated and compatibility is no longer needed.
       await this.notificationServiceV2.deleteSubscription({
         deviceUuid: uuid,
         chainId: chainId,
         safeAddress: safeAddress,
       });
+    } catch (error: unknown) {
+      if (error instanceof NotFoundException) {
+        // Do not throw a NotFound error when attempting to remove the token from the CGW,
+        // This ensures the TX service remove method is called
+      } else {
+        throw error;
+      }
+    }
+
+    try {
+      await this.notificationsService.unregisterSafe({
+        chainId,
+        uuid,
+        safeAddress,
+      });
+    } catch (error: unknown) {
+      // The token might already have been removed from the TX service.
+      // If this happens, the TX service will throw a 404 error, but it is safe to ignore it.
+      if (error instanceof Error) {
+        if ('code' in error && error.code !== 404) {
+          throw error;
+        }
+      }
```

Comment:
If the error is not `instanceof Error`, it is being swallowed here.

Reply:
Change in the new version
https://github.com/safe-global/safe-client-gateway/pull/2173/commits/6a9aa2f0f392ae47d6632dcfe27a6851fdcbe505